src/register/siselect.rs

```diff
@@ -0,0 +1,156 @@
+//! siselect register
+//!
+//! The `siselect` CSR is defined in "The RISC-V Advanced Interrupt
+//! Architecture" Version 0.3.2-draft
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Register {
+    Eidelivery = 0x70,
+    Eithreshold = 0x72,
+    Eip0 = 0x80,
```

Comment:
nit: You already have the enums all instantiated, so its okay to keep it this way. But, with the recent update, you don't really need to define all Eip* and Eie* register addresses. There can be a single EipBase, EieBase and then calculate the required address from that.